packages/admin/components/forms/ParentSelector/ParentSelector.tsx

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import { graphql } from "react-relay";
+import { DialogDisclosure, useDialogState } from "reakit/Dialog";
+import { useTranslation } from "react-i18next";
+import * as Styled from "./ParentSelector.styles";
+import ParentSelectorModal from "./ParentSelectorModal";
+import { useMaybeFragment } from "hooks";
+import BaseInputLabel from "components/forms/BaseInputLabel";
+import { ButtonControl } from "components/atomic";
+
+import { ParentSelectorFragment$key } from "@/relay/ParentSelectorFragment.graphql";
+
+const ParentSelector = ({ data }: Props) => {
+  const entity = useMaybeFragment(fragment, data);
+  const parentFragment =
+    entity?.__typename === "Collection"
+      ? collectionParentFragment
+      : itemParentFragment;
+  const parent = useMaybeFragment(parentFragment, entity?.parent);
```

Comment:
@jen-castiron I couldn't figure out the type error here. I think I must be missing something simple. I tried splitting out the parent fragments to fix it, and that didn't work. Once this is resolved, this PR should be ready to go.